src/tests/ftest/util/fio_utils.py

```diff
@@ -0,0 +1,331 @@
+        self.disable_slat = FormattedParameter("--disable_slat={}")
+        self.disable_bw_measurement = FormattedParameter(
+            "--disable_bw_measurement={}")
+        self.gtod_cpu = FormattedParameter("--gtod_cpu={}")
+        self.unified_rw_reporting = FormattedParameter(
+            "--unified_rw_reporting={}")
+        self.continue_on_error = FormattedParameter("--continue_on_error={}")
+        self.error_dump = FormattedParameter("--error_dump={}")
+        self.profile = FormattedParameter("--profile={}")
+        self.cgroup = FormattedParameter("--cgroup={}")
+        self.cgroup_nodelete = FormattedParameter("--cgroup_nodelete={}")
+        self.cgroup_weight = FormattedParameter("--cgroup_weight={}")
+        self.uid = FormattedParameter("--uid={}")
+        self.gid = FormattedParameter("--gid={}")
+        self.kb_base = FormattedParameter("--kb_base={}")
+        self.unit_base = FormattedParameter("--unit_base={}")
+        self.hugepage_size = FormattedParameter("--hugepage-size={}")
+        self.flow_id = FormattedParameter("--flow_id={}")
+        self.flow = FormattedParameter("--flow={}")
+        self.flow_watermark = FormattedParameter("--flow_watermark={}")
+        self.flow_sleep = FormattedParameter("--flow_sleep={}")
+        self.steadystate = FormattedParameter("--steadystate={}")
+        self.steadystate_duration = FormattedParameter(
+            "--steadystate_duration={}")
+        self.steadystate_ramp_time = FormattedParameter(
+            "--steadystate_ramp_time={}")
+
+        # fio api
+        self.api = BasicParameter(None, "POSIX")
+
+    def get_param_names(self):
+        """Get a sorted list of the defined FioCommand parameters."""
+        # Sort the FIO parameter names to generate consistent fio commands
+        all_param_names = super(FioCommand, self).get_param_names()
+        # move the 'name' option as the first option of fio job
+        all_param_names.remove("name")
+        all_param_names.insert(0, "name")
+
+        return all_param_names
+
+
+class Fio(FioCommand):
+    """Class defining an object of type FioCommand"""
+
+    def __init__(self, namespace, test):
+        """Create a fio object"""
+        # Get run command by iterating over different fio jobs
+        # mentioned in yaml file under '/run/fio/'.
+        self.run_cmd = ''
+        for name in namespace:
+            super(Fio, self).__init__(name, "fio", test)
+            self.get_params(test)
+            self.run_cmd += self.__str__()
+
+    def run(self, host):
+        """Execute Fio from the host passed on to the method.
+
+           Args:
+             host: hostname from where to launch fio
+        """
+        # run fio command
+        self.run_cmd = self.run_cmd.replace("fio", ' ').replace(" POSIX", '')
+        ret_code = general_utils.pcmd(host, 'fio' + self.run_cmd)
+
+        # check for any failures
+        if 0 not in ret_code:
```

Comment:
It is best to convert the ret_code to integer and check "if not ret_code". There are non-zero values that have "0" in them which will skip this block while it shouldn't.

Reply:
The `ret_code` object is a dictionary of integer keys with `NodeSet` values (the list of hosts that yielded the return code that matches the key).  So `<int> not in <dict>` where the keys are of type `int` is the efficient way to do this.

Reply:
Agreed. Initially I thought ret_code has single string value. Later, Saurabh explained me that it was a dict and it makes sense to keep the code this way.